lib/preface_specs/bounded_meet_semilattice.mli

```diff
@@ -0,0 +1,59 @@
+(** A [Bounded_meet_semilattice] capture the notion of a set with
+    {b greatest lower bound} with a top. A bounded meet semilattice is an
+    idempotent commutative semigroup.
+
+    {2 Laws}
+
+    To have a predictable behaviour, the instance of [Bounded_meet_semilattice]
+    must obey some laws.
+
+    - [meet a (meet b c) = meet (meet a b) c]
+    - [meet a b = meet b a]
+    - [meet a a = a]
+    - [meet a top = a] *)
+
+(** {1 Minimal definition}*)
+
+module type WITH_MEET = sig
+  type t
+  (** The type held by the [Bounded_meet_semilattice] *)
+
+  include Meet_semilattice.WITH_MEET with type t := t
+  (** @inline *)
+
+  val top : t
+  (** the top value represent the greatest element of the meet semilattice *)
+end
+
+(** {1 Structure anatomy} *)
+
+module type CORE = WITH_MEET
+(** Basis operations. *)
+
+(** Infix operators. *)
+module type INFIX = sig
```

Comment:
But here, you make the right choice since we want to propagate the documentation properly. (You can resolve this comment)